L1Trigger/L1TMuonEndCapPhase2/interface/EMTFContext.h

```diff
@@ -0,0 +1,72 @@
+#ifndef L1Trigger_L1TMuonEndCapPhase2_EMTFContext_h
+#define L1Trigger_L1TMuonEndCapPhase2_EMTFContext_h
+
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFfwd.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFConfiguration.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFModel.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFTypes.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/HitmapLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/PatternMatchingLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/RoadSortingLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/TrackBuildingLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/DuplicateRemovalLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/ParameterAssignmentLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Algo/OutputLayer.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/ActivationLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/HostLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/SiteLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/ZoneLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/TimeZoneLut.h"
+
+namespace emtf::phase2 {
+
+  // Class
+  class EMTFContext {
+  public:
+    EMTFContext(const edm::ParameterSet&, edm::ConsumesCollector);
+
+    ~EMTFContext();
+
+    // Event configuration
+    void update(const edm::Event&, const edm::EventSetup&);
+
+    // Parameter Set
+    const edm::ParameterSet& pset_;
```

Comment:
Does the constructor argument `ParameterSet` originate from the `ParameterSet` argument of an EDModule constructor? If that is the case, you must not store a reference to it (that `ParameterSet` object will be deleted after `beginJob` transition).

In general it would be best to get the necessary parameters from `ParameterSet` early and store them in some compact form. The `ParameterSet` is a heavy way of storing numbers, and the `get()` function is relatively expensive as well.